src/kbmod/result_set.py

```diff
@@ -0,0 +1,521 @@
+        self._final_likelihood = trj.lh
+        self._all_times = copy.copy(times)
+        self._valid_indices = [i for i in range(len(times))]
+        self._all_stamps = None
+        self._psi_curve = None
+        self._phi_curve = None
+        self._num_times = len(times)
+
+    @property
+    def trajectory(self):
+        return self._trajectory
+
+    @property
+    def stamp(self):
+        return self._stamp
+
+    @property
+    def final_lh(self):
+        return self._final_likelihood
+
+    @property
+    def final_likelihood(self):
+        return self._final_likelihood
+
+    @property
+    def times(self):
+        return self._all_times
+
+    @property
+    def valid_indices(self):
+        return self._valid_indices
+
+    @property
+    def all_stamps(self):
+        return self._all_stamps
+
+    @property
+    def psi_curve(self):
+        return self._psi_curve
+
+    @property
+    def phi_curve(self):
+        return self._phi_curve
+
+    def set_stamp(self, stamp):
+        self._stamp = stamp
+
+    def set_all_stamps(self, all_stamps):
+        assert(len(all_stamps) <= self._num_times)
+        self._all_stamps = all_stamps
+
+    def set_psi_phi(self, psi, phi):
+        """
+        Set the psi and phi curves and auto calculate the light curve.
+        
+        Arguments:
+            psi : List - The psi curve
+            phi : List - The phi curve
+        """
+        assert(len(psi) == len(phi))
+        assert(len(psi) <= self._num_times)
+        self._psi_curve = psi
+        self._phi_curve = phi
+        self._update_likelihood()
+
+    def get_valid_times(self):
+        return [self._all_times[i] for i in self._valid_indices]
+
+    def get_trj_result(self):
+        """
+        Return the current trajectory information as a trj_result object.
+        """
+        return kb.trj_result(self._trajectory, self._num_times, self._valid_indices)
+
+    def filter_indices(self, indices_to_keep):
+        """
+        Remove invalid indices and times from the ResultDataRow. This uses relative filtering
+        where valid_indices[i] is kept for all i in indices_to_keep.
+        
+        Arguments:
+            indices_to_keep : List of ints - which indices to keep.
+        """
+        current_num_inds = len(self._valid_indices)
+        assert(not any(v >= current_num_inds or v < 0 for v in indices_to_keep))
+
+        self._valid_indices = [self._valid_indices[i] for i in indices_to_keep]
+        self._update_likelihood()
+
+    def compute_light_curve(self):
+        """
+        Fill the light curve from the psi and phi curves.
+        """
+        if self._psi_curve is None or self._phi_curve is None:
+            return []
+
+        num_elements = len(self._psi_curve)
+        lc = [0.0] * num_elements
+        for i in range(num_elements):
+            if self._phi_curve[i] != 0.0:
+                lc[i] = self._psi_curve[i] / self._phi_curve[i]
+        return lc
```

Comment:
I've been chatting with Andy to try and figure out how important would it be for the `ResultSet` to be fast and he came up with the following calculation for DEEP: 
```
10 pointing groups per region (e.g. B1)
~4 regions
60 ccds per pointing group
10-100 candidates per CCD
so 24k - 240k
```
This is after filtering duplicates etc. So these would be in the daily-use result set. On the upper scale, that would be several GBs of results, which makes me think it would be somewhat important to not waste too much time. Definitely worth writing a mock example for.